solr/core/src/java/org/apache/solr/core/JavaagentTracerConfigurator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.core;
+
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+import io.opentracing.util.GlobalTracer;
+import org.apache.solr.common.util.NamedList;
+
+public class JavaagentTracerConfigurator extends TracerConfigurator {
```

Comment:
Can you add a couple of lines of javadoc describing what this class does? I am guessing the GlobalTracer is the java agent tracer?

Reply:
In previous commit, there are no way to configure sampleRate without declaring it in `Solr.xml` with a proper implementation of `TracerConfigurator`. Therefore `JavaagentTracerConfigurator` was introduced.

Since sampleRate is moved to cluster properties now, `TracerConfigurator` can do the job of `JavaagentTracerConfigurator` now.